server.go

```diff
@@ -411,30 +411,23 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq, inbound
 
 	// Now that we've established a connection, create a peer, and
 	// it to the set of currently active peers.
-	peer, err := newPeer(conn, s, peerAddr, false)
+	p, err := newPeer(conn, connReq, s, peerAddr, inbound)
 	if err != nil {
 		srvrLog.Errorf("unable to create peer %v", err)
-		s.connMgr.Remove(peer.connReq.ID())
-		peer.Disconnect()
-		conn.Close()
+		p.Disconnect()
```

Comment:
The diff on this first commit can be made much more minimal: 
 * The root cause is this line `s.connMgr.Remove(peer.connReq.ID())`

As a result, all the needs to be done to fix this issue is check that the `peer.connReq` isn't nil before attempting to access it. The `Remove` there is _required_ as it tells the `connmgr` *not* to attempt to retry the connection attempts. 

Can you modify the first commit to only have this conditional `connmgr.Remove`, both here and below? Thanks!